server/test/support/ResourceCreator.java

```diff
@@ -120,6 +124,7 @@ public Applicant insertApplicantWithAccount(Optional<String> accountEmail) {
     Account account = insertAccount();
 
     accountEmail.ifPresent(account::setEmailAddress);
+    account.save();
```

Comment:
Note to the reviewer: Not calling this before meant that the call to setEmailAddress above was ineffective.